app/models/bishop.rb

```diff
@@ -8,5 +8,21 @@ def icon
       '&#9821;'
     end
   end
+  
+  def rook_valid_move?(to_x, to_y)
+    return false if self.is_obstructed(to_x, to_y)
+    rook.horizantal_move?(to_y) || rook.vertical_move?(to_x)
+  end
 
-end
\ No newline at end of file
+  private
+  
+  def horizantal_move?(to_y)
+    if to_y != starting_point_y
+    end
+  end
+  
+  def vertical_move?(to_x)
+    if to_y != starting_point_x
```

Comment:
I must've written the test in both files. It was a late night, lol. I deleted it.